dds/DCPS/transport/framework/TransportRegistry.cpp

```diff
@@ -782,31 +820,31 @@ TransportRegistry::config_has_transport_template(const ACE_TString& config_name)
 }
 
 bool
-TransportRegistry::get_transport_template_info(const OPENDDS_STRING& config_name, TransportTemplate& inst)
+TransportRegistry::get_transport_template_info(const ACE_TString& config_name, TransportTemplate& inst)
 {
   bool ret = false;
   if (has_transport_template()) {
     for (OPENDDS_VECTOR(TransportTemplate)::const_iterator i = transport_templates_.begin(); i != transport_templates_.end(); ++i) {
-      if (!ACE_OS::strcmp(config_name.c_str(), i->config_name.c_str())) {
+      if (!ACE_OS::strcmp(ACE_TEXT_ALWAYS_CHAR(config_name.c_str()), i->config_name.c_str())) {
         inst.transport_template_name = i->transport_template_name;
         inst.config_name = i->config_name;
         inst.instantiate_per_participant = i->instantiate_per_participant;
         inst.customizations = i->customizations;
         inst.transport_info = i->transport_info;
 
         ret = true;
-
-        if (DCPS_debug_level > 0) {
-          ACE_DEBUG((LM_DEBUG,
-                     ACE_TEXT("(%P|%t) TransportRegistry::get_transport_template_info: ")
-                     ACE_TEXT("found config %C\n"),
-                     config_name.c_str()));
-        }
-
         break;
       }
     }
   }
+
+  if (DCPS_debug_level > 0) {
+    ACE_DEBUG((LM_DEBUG,
+               ACE_TEXT("(%P|%t) TransportRegistry::get_transport_template_info: ")
+               ACE_TEXT("%C config %C\n"),
+               ret ? "found" : "did not find", config_name.c_str()));
```

Comment:
Here, I think you want %s for config_name.c_str() now since TString::c_str() is an ACE_TCHAR*.